server/src/handler.ts

```diff
@@ -133,10 +137,32 @@ export class Handler {
       }
     }
 
+    await this.addDependencies(this.reg);
+
     progress.report(0, "Parsing files");
     await this.reg.parseAsync(new Progress(progress));
   }
 
+  private async addDependencies(reg: abaplint.Registry) {
+    const deps = this.reg.getConfig().get().dependencies;
+    if (deps) {
+      deps.forEach(function (dep) {
+        (async () => {
+          process.stderr.write("Clone: " + dep.url + "\n");
+          const dir = fs.mkdtempSync(path.join(os.tmpdir(), "abaplint-"));
+          childProcess.execSync("git clone --quiet --depth 1 " + dep.url + " .", {cwd: dir, stdio: "inherit"});
+          const names = FileOperations.loadFileNames(dir + dep.files);
+          let files: abaplint.IFile[] = [];
+          files = files.concat(await FileOperations.loadFiles(false, names));
+          files.forEach(function (file) {
+            reg.addFile(new abaplint.MemoryFile(file.getFilename(), file.getRaw()));
+          });
+          FileOperations.deleteFolderRecursive(dir);
+        })();
+      });
+    }
+  }
```

Comment:
can leave out the function argument if you use arrow functions instead of "function" keyword:
```typescript
private async addDependencies() {
    const deps = this.reg.getConfig().get().dependencies;
    if (deps) {
      deps.forEach((dep) => {
        (async () => {
          process.stderr.write("Clone: " + dep.url + "\n");
          const dir = fs.mkdtempSync(path.join(os.tmpdir(), "abaplint-"));
          childProcess.execSync("git clone --quiet --depth 1 " + dep.url + " .", {cwd: dir, stdio: "inherit"});
          const names = FileOperations.loadFileNames(dir + dep.files);
          let files: abaplint.IFile[] = [];
          files = files.concat(await FileOperations.loadFiles(false, names));
          files.forEach((file) => {
            this.reg.addFile(new abaplint.MemoryFile(file.getFilename(), file.getRaw()));
          });
          FileOperations.deleteFolderRecursive(dir);
        })();
      });
    }
  }
```

Reply:
yes, I changed it to the fat arrow

Reply:
thanks, now you can leave out the registry as a function argument, instead:
```typescript
private async addDependencies() 
```
and
```typescript
     await this.addDependencies(); 
```

Reply:
Hopefully I get you right :).
I deleted the async/await.

Reply:
now I get it. I removed the registry as a function argument

Reply:
added async/await again with leave out the registry as a function argument.